### PR TITLE
Preventing to add host twice for single node deployment

### PR DESCRIPTION
**Pull request by gobindadas**
Bug-Url: https://bugzilla.redhat.com/show_bug.cgi?id=1715959

### CODE REVIEW

**Comment by ovirt-infra** [comment]
Hello contributor, thanks for submitting a PR for this project! 

I am the bot who triggers "standard-CI" builds for this project.
As a security measure, I will not run automated tests on PRs that are not from white-listed contributors.

In order to allow automated tests to run, please ask one of the project maintainers to review the code and then do one of the following:

1. Type `ci test please` on this PR to trigger automated tests for it.
2. Type `ci add to whitelist` on this PR to trigger automated tests for it and also add you to the contributor white-list so that your future PRs will be tested automatically. ( keep in mind this list might be overwritten if the job XML is refreshed, for permanent whitelisting, please follow #3 option )
3. If you are planning to contribute to more than one project, maybe it's better to ask them to add you to the project organization, so you'll be able to run tests for all the organization's projects.

**Comment by gobindadas** [comment]
@tiraboschi Can you please review and merge? This needs to go with 4.3.6

File: hooks/after_setup/add_host_storage_domain.yml [hunk 1]
```diff
@@ -33,7 +33,7 @@
   with_items: "{{ gluster.hosts }}"
   loop_control:
     loop_var: host
-  when: "gluster is defined and 'hosts' in gluster"
+  when: gluster is defined and "hosts in gluster" and gluster.hosts | length > 1
```
Comment (tiraboschi) [inline hooks/after_setup/add_host_storage_domain.yml]: why do you need the double quotes around "hosts in gluster"?

Comment (gobindadas) [reply hooks/after_setup/add_host_storage_domain.yml]: Removed

**Pull request merged**